applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -0,0 +1,435 @@
+        {
+            subspace_size = 2 * static_cast<size_t>(mParam["number_of_eigenvalues"].GetInt());
+        }
+        else if( mParam["subspace_size"].GetInt() == 0 )
+        {
+            subspace_size = 1.5 * static_cast<size_t>(mParam["number_of_eigenvalues"].GetInt());
+        }
+        else
+        {
+            subspace_size = static_cast<size_t>(mParam["subspace_size"].GetInt());
+        }
+
+        // create column based matrix for the fortran routine
+        FEASTMatrixType tmp_eigenvectors(system_size, subspace_size);
+        OutputVectorType tmp_eigenvalues(subspace_size);
+        VectorType residual(subspace_size);
+
+        // set FEAST settings
+        int fpm[64] = {};
+        feastinit(fpm);
+
+        mParam["echo_level"].GetInt() > 0 ? fpm[0] = 1 : fpm[0] = 0;
+
+        fpm[2] = -std::log10(mParam["tolerance"].GetDouble());
+        fpm[3] = mParam["max_iteration"].GetInt();
+
+        // compute only right eigenvectors
+        if( !TSymmetric )
+            fpm[14] = 1;
+
+        if( mParam["search_lowest_eigenvalues"].GetBool() )
+        {
+            fpm[39] = -1;
+        }
+        if( mParam["search_highest_eigenvalues"].GetBool() )
+        {
+            fpm[39] = 1;
+        }
+
+        char UPLO = 'F';
+        int N = static_cast<int>(system_size);
+
+        // provide matrices in array form. fortran indices start with 1, must be int
+        double* A = reinterpret_cast<double*>(rK.value_data().begin());
+        std::vector<int> IA;
+        IA.reserve(N+1);
+        for( int i=0; i<N+1; ++i )
+        {
+            IA[i] = static_cast<int>(rK.index1_data()[i]) + 1;
+        }
+        std::vector<int> JA;
+        JA.reserve(IA[N]-1);
+        for( int i=0; i<IA[N]-1; ++i )
+        {
+            JA[i] = static_cast<int>(rK.index2_data()[i]) + 1;
+        }
+
+        double* B = reinterpret_cast<double*>(rM.value_data().begin());
+        std::vector<int> IB;
+        IB.reserve(N+1);
+        for( int i=0; i<N+1; ++i )
+        {
+            IB[i] = static_cast<int>(rM.index1_data()[i]) + 1;
+        }
+        std::vector<int> JB;
+        JB.reserve(IB[N]-1);
+        for( int i=0; i<IB[N]-1; ++i )
+        {
+            JB[i] = static_cast<int>(rM.index2_data()[i]) + 1;
+        }
+
+        double epsout;
+        int loop;
+        TScalarOut E1 = GetE1(T);
+        double E2 = GetE2(T);
+        double* Emin = reinterpret_cast<double*>(&E1);
+        double* Emax = reinterpret_cast<double*>(&E2);
+        int M0 = static_cast<int>(subspace_size);
+        double* E = reinterpret_cast<double*>(tmp_eigenvalues.data().begin());
+        double* X = reinterpret_cast<double*>(tmp_eigenvectors.data().begin());
+        int M;
+        double* res = reinterpret_cast<double*>(residual.data().begin());
+        int info;
+
+        // call feast
+        auto feast = CallFeast(Ti, TSymmetric);
```

Comment:
```suggestion
        auto feast = CreateFeast(Ti, TSymmetric);
```

Looks more like this

Reply:
ok